packages/actions/test/e2e/01-setup.test.ts

```diff
@@ -1,58 +1,225 @@
-import chai from "chai"
+import chai, { expect, assert } from "chai"
 import chaiAsPromised from "chai-as-promised"
+import { randomBytes } from "crypto"
+import fs from "fs"
+import { getAuth, signInWithEmailAndPassword } from "firebase/auth"
+import {
+    initializeAdminServices,
+    initializeUserServices,
+    getStorageConfiguration,
+    generatePseudoRandomStringOfNumbers,
+    createNewFirebaseUserWithEmailAndPw,
+    deleteAdminApp,
+    sleep,
+    setCustomClaims,
+    deleteBucket,
+    deleteObjectFromS3,
+    envType
+} from "../utils"
+import { fakeCeremoniesData, fakeCircuitsData, fakeUsersData } from "../data/samples"
+import {
+    getBucketName,
+    createS3Bucket,
+    getCurrentFirebaseAuthUser,
+    multiPartUpload,
+    getZkeyStorageFilePath,
+    getPotStorageFilePath,
+    getR1csStorageFilePath,
+    setupCeremony
+} from "../../src"
+import { TestingEnvironment } from "../../src/types/enums"
 
 // Config chai.
 chai.use(chaiAsPromised)
 
 describe("Setup", () => {
```

Comment:
It would appear that only the ceremony creation scenario has been tested successfully.

I was wondering, what other scenarios would be worth testing for the V1 release? Perhaps none, but I would put more attention on the output produced at the end of the workflow of the tested scenario (double check the checks and add what's missing, we can't go wrong there).

@gurrpi @daodesigner Do you get any ideas about possible scenarios to test?

Reply:
Order of importance for tests is probably: contribution >> finalize > set up; most important thing to test on set up side is probably that resources are locked down as tight as possible: can't override another's ceremony, fails gracefully without affecting other ceremonies (ie. if circom doesn't compile), can't XSS, etc.

Reply:
We already implemented some good protections on collections data (users) and S3 in general (cannot get pre-signed urls for objects not part of a ceremony -> which are not sensitive) needs a bit more work, although it will also boil down to how the coordinator sets up the IAM role for accessing S3. 
Input validation tests will be added as I get properly down with pentesting this.

Reply:
> Order of importance for tests is probably: contribution >> finalize > set up; most important thing to test on set up side is probably that resources are locked down as tight as possible: can't override another's ceremony, fails gracefully without affecting other ceremonies (ie. if circom doesn't compile), can't XSS, etc.

Strongly agree w/ the order of importance and thanks @daodesigner for to always pull something interesting out of the hat :heart_decoration: 

> We already implemented some good protections on collections data (users) and S3 in general (cannot get pre-signed urls for objects not part of a ceremony -> which are not sensitive) needs a bit more work, although it will also boil down to how the coordinator sets up the IAM role for accessing S3. Input validation tests will be added as I get properly down with pentesting this.

Thanks @ctrlc03 for the quick and complete answer. Let's add the test case on ceremony override (S3 should avoid creating the same bucket by default and an error should be triggered by CF). Regarding gracefully failure, each ceremony is independent in terms of bucket + collection space and has no relation to other ceremonies spaces. Therefore, if something goes wrong on the ceremony setup side, we just need to make sure to delete the intermediate/incomplete data from the Firestore DB and remove the already created S3 bucket (if any).